File: admin-web/src/types/showtime.ts
import type { Movie } from "./movie";
import type { Room } from "./room";

export type ShowtimeStatus = "OPEN" | "CANCELLED";

export type Showtime = {
  id: string;
  movieId: string;
  roomId: string;
  startTime: string;
  endTime: string;
  basePrice: number;
  status: ShowtimeStatus;
  movie?: Movie;
  room?: Room;
};

export type ShowtimePayload = {
  movieId: string;
  roomId: string;
  startTime: string;
  endTime?: string;
  basePrice: number;
  status?: ShowtimeStatus;
};

File: client-app/app/src/main/java/com/uit/eousx/domain/model/Showtime.kt
package com.uit.eousx.domain.model

data class Showtime(
    val id: String,
    val movieId: String,
    val movieTitle: String,
    val roomId: String,
    val roomName: String,
    val roomType: String,
    val startTime: String,
    val endTime: String,
    val basePrice: Int,
    val status: String,
    val availableSeats: Int
)
